Ginger/GingerCore/ALM/QtestCore.cs

```diff
@@ -0,0 +1,685 @@
+            foreach (QTestApiModel.TestRunWithCustomFieldResource testRun in testRunList)
+            {
+                QtestTest test = GetQtestTest(testRun.TestCase.Id);
+                test.Runs = new List<QtestTestRun>();
+                test.Runs.Add(new QtestTestRun(testRun.Id.ToString(), testRun.Name, testRun.Properties[0].ToString(), testRun.CreatorId.ToString()));
+                TS.Tests.Add(test);
+            }
+          
+            return TS;
+        }
+
+        public ObservableList<QTestApiModel.TestCycleResource> GetQTestCyclesByProject(string qTestServerUrl, string qTestUserName, string qTestPassword, string qTestProject)
+        {
+            ConnectALMServer();
+            ObservableList<QTestApiModel.TestCycleResource> cyclestList = new ObservableList<QTestApiModel.TestCycleResource>();
+            QTestApi.TestcycleApi TestcycleApi = new QTestApi.TestcycleApi(connObj.Configuration);
+            cyclestList = new ObservableList<QTestApiModel.TestCycleResource>(TestcycleApi.GetTestCycles((long)Convert.ToInt32(qTestProject), null, null, "descendants"));
+
+            return cyclestList;
+        }
+
+        public void UpdatedQCTestInBF(ref BusinessFlow businessFlow, List<QtestTest> tcsList)
+        {
+            ImportFromQtest.UpdatedQCTestInBF(ref businessFlow, tcsList);
+        }
+
+        public void UpdateBusinessFlow(ref BusinessFlow businessFlow, List<QtestTest> tcsList)
+        {
+            ImportFromQtest.UpdateBusinessFlow(ref businessFlow, tcsList);
+        }
+
+        public BusinessFlow ConvertQCTestSetToBF(QtestTestSuite TS)
+        {
+            return ImportFromQtest.ConvertQtestTestSuiteToBF(TS);
+        }
+
+        public QtestTestSuite GetQtestTestSuite(string testSuiteID)
+        {
+            QtestTestSuite testSuite = new QtestTestSuite();
+            testSuite.ID = testSuiteID;
+            return ImportTestSetData(testSuite);          
+        }
+
+        public override Dictionary<Guid, string> CreateNewALMDefects(Dictionary<Guid, Dictionary<string, string>> defectsForOpening, List<ExternalItemFieldBase> defectsFields, bool useREST)
+        {
+            if (!useREST)
+            {
+                // do nothing
+                return new Dictionary<Guid, string>();
+            }
+            else
+            {
+                return ImportFromQC.CreateNewDefectQCREST(defectsForOpening);
+            }
+        }
+
+        public static string StripHTML(string HTMLText, bool toDecodeHTML = true)
+        {
+            try
+            {
+                HTMLText = HTMLText.Replace("<br />", Environment.NewLine);
+                Regex reg = new Regex("<[^>]+>", RegexOptions.IgnoreCase);
+                var stripped = reg.Replace(HTMLText, "");
+                if (toDecodeHTML)
+                    stripped = HttpUtility.HtmlDecode(stripped);
+
+                stripped = stripped.TrimStart(new char[] { '\r', '\n' });
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the array type; it is redundant.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5395796)